swig/include/python/gdal_python.i

```diff
@@ -352,6 +352,44 @@ void wrapper_VSIGetMemFileBuffer(const char *utf8_path, GByte **out, vsi_l_offse
 %clear (GByte **out, vsi_l_offset *length);
 
 
+%pythonappend VSIFCloseL %{
+    args[0].this = None
+%}
+
+%pythonprepend VSIFCloseL %{
```

Comment:
These `%pythonprepend` blocks are not necessary to avoid the crash, they just make it raise the standard python exception instead of e.g. `TypeError: in method 'VSIFEofL', argument 1 of type 'VSILFILE *'
`